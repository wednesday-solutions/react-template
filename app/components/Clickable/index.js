/**
 *
 * Clickable
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import T from '@components/T';
import { text } from '@app/themes/colors';

const StyledClickable = styled.div`
  color: ${text};
  &:hover {
    cursor: pointer;
  }
`;

/**
 * Clickable component that renders a styled element which can be clicked.
 * Optionally displays text identified by a text ID.
 *
 * @date 01/03/2024 - 14:47:28
 *
 * @param {Object} props - The component props.
 * @param {Function} props.onClick - The function to call when the element is clicked.
 * @param {string} [props.textId] - The ID for the text to display inside the clickable element.
 * @returns {JSX.Element} A styled clickable element, optionally containing text.
 */
export function Clickable({ onClick, textId }) {
  return (
    <StyledClickable data-testid="clickable" onClick={onClick}>
      {textId && <T id={textId} />}
    </StyledClickable>
  );
}

Clickable.propTypes = {
  onClick: PropTypes.func.isRequired,
  textId: PropTypes.string.isRequired
};

export default Clickable;
