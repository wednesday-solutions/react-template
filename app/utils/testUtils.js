import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter, Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import configureStore from '@app/configureStore';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@containers/LanguageProvider';
import { IntlGlobalProvider } from '@components/IntlGlobalProvider';
import { theme } from '@containers/App';

export const renderWithIntl = (children) =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );

export const renderProvider = (children, history) => {
  const store = configureStore({}, browserHistory).store;
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        <ThemeProvider theme={theme}>
          {history ? <Router history={history}>{children}</Router> : <BrowserRouter>{children}</BrowserRouter>}
        </ThemeProvider>
      </ConnectedLanguageProvider>
    </Provider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const getStyles = () => {};
