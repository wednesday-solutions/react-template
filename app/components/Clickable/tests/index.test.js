/**
 *
 * Tests for Clickable
 *
 */

import React from 'react';
import { fireEvent } from '@testing-library/react';
import { renderWithIntl } from '@utils/testUtils';
import Clickable from '../index';

describe('<Clickable /> component tests', () => {
  let clickSpy;
  beforeAll(() => {
    clickSpy = jest.fn();
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Clickable textId="repo_list" onClick={clickSpy} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Clickable component', () => {
    const { getAllByTestId } = renderWithIntl(<Clickable textId="repo_list" onClick={clickSpy} />);
    expect(getAllByTestId('clickable').length).toBe(1);
  });

  it('should contain render the text according to the textId', () => {
    const { getAllByText } = renderWithIntl(<Clickable textId="repo_list" onClick={clickSpy} />);
    expect(getAllByText(/Repository List/).length).toBe(1);
  });

  it('should call the prop onClick when the clickable component is clicked', () => {
    const { getAllByText, queryByText } = renderWithIntl(<Clickable onClick={clickSpy} textId="repo_list" />);
    expect(getAllByText(/Repository List/).length).toBe(1);
    fireEvent.click(queryByText(/Repository List/));
    expect(clickSpy).toBeCalled();
  });
});
