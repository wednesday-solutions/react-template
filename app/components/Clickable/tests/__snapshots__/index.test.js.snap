// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clickable /> component tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="Clickable__StyledClickable-sc-19lp5om-0 avquq"
      data-testid="clickable"
    />
  </div>
</body>
`;
